pre_commit/clientlib/validate_config.py

```diff
@@ -52,8 +59,19 @@ def try_regex(repo, hook, value, field_name):
 
 
 def validate_config_extra(config):
+    local_repos_found = False
     for repo in config:
-        if 'sha' not in repo:
+        if is_local_hooks(repo):
+            if local_repos_found:
+                raise InvalidConfigError(
+                    'Multiple local hooks found'
+                )
```

Comment:
I feel that if it's not broken, why go through the extra trouble to exclude the case?

Also, since hooks run in the order they are defined, it might be desirable to run some hooks out-of-order.  We currently don't forbid repeated repos at the moment so I see no reason to make `local` repos special in that regard.